### PR TITLE
cnf network: add rdma cni test cases

### PR DESCRIPTION
**Pull request by ajaggapa**
`rdmametricsapi.go`
Test Cases for 4.18 RDMA CNI feature with SriovNetworkPoolConfig API

`rdmametrics.go`
Removed this file

### CODE REVIEW

File: tests/cnf/core/network/sriov/tests/rdmametricsapi.go [hunk 1]
```diff
@@ -193,51 +200,94 @@ var _ = Describe("rdmaMetrics", Ordered, Label(tsparams.LabelRdmaMetricsTestCase
...
+
+	var allNodes []string
+
+	for _, node := range nodelist {
+		allNodes = append(allNodes, node.Object.Name)
 	}
 
-	finalIgnitionConfig, err := json.Marshal(ignitionConfig)
-	Expect(err).ToNot(HaveOccurred(), "Failed to serialize ignition config")
+	sriovNodeState, err := sriov.ListNetworkNodeState(APIClient, NetConfig.SriovOperatorNamespace)
```
Comment (evgenLevin) [inline tests/cnf/core/network/sriov/tests/rdmametricsapi.go:236]:
```suggestion
	sriovNodeStates, err := sriov.ListNetworkNodeState(APIClient, NetConfig.SriovOperatorNamespace)
```

File: tests/cnf/core/network/sriov/tests/rdmametricsapi.go [hunk 2]
```diff
@@ -193,51 +200,94 @@ var _ = Describe("rdmaMetrics", Ordered, Label(tsparams.LabelRdmaMetricsTestCase
...
+	Expect(err).ToNot(HaveOccurred(), "Failed to fetch SriovNetworkNodeStates")
 
-	createdMC, err := mco.NewMCBuilder(APIClient, "02-rdma-netns-exclusive-mode").
-		WithLabel("machineconfiguration.openshift.io/role", NetConfig.CnfMcpLabel).
-		WithRawConfig(finalIgnitionConfig).
-		Create()
-	Expect(err).ToNot(HaveOccurred(), "Failed to create RDMA machine config")
+	for _, node := range sriovNodeState {
```
Comment (evgenLevin) [inline tests/cnf/core/network/sriov/tests/rdmametricsapi.go:239]:
```suggestion
	for _, nodeState := range sriovNodeStates {
```

**Review by evgenLevin — COMMENTED** [review]
Minnor comments

File: tests/cnf/core/network/sriov/tests/rdmametricsapi.go (hunk 1 above)
Comment (ajaggapa) [reply tests/cnf/core/network/sriov/tests/rdmametricsapi.go]: updated

File: tests/cnf/core/network/sriov/tests/rdmametricsapi.go (hunk 2 above)
Comment (ajaggapa) [reply tests/cnf/core/network/sriov/tests/rdmametricsapi.go]: updated

**Review by gkopels — COMMENTED** [review]
One very small comment

**Review by gkopels — APPROVED** [review]

**Comment by gkopels** [comment]
lgtm

**Review by gkopels — APPROVED** [review]

**Pull request merged**